days/src/day7/Day7.kt

```diff
@@ -0,0 +1,66 @@
+package day7
+
+import Day
+import java.io.File
+
+class Day7(private val bagRules: Map<String, List<Pair<Int, String>>>) : Day {
+    private lateinit var containmentGraph: Map<String, List<String>>
+
+    init {
+        val graph: MutableMap<String, MutableList<String>> = mutableMapOf()
+        bagRules.forEach { (containingBag, contained) ->
+            contained.forEach { (amount, bag) ->
+                val newEdgeList = graph.getOrDefault(bag, mutableListOf())
+                newEdgeList.add(containingBag)
+                graph[bag] = newEdgeList
```

Comment:
Is the only reason you didn't compress this into a single line for readability?

Reply:
I was debugging, haha